### DISCUSSION TITLE
Gathering feedback on recent arrow changes (async primitives to lib device abstraction in arrow c++): is this/would this be useful to Daft?

### DISCUSSION THREAD

**Discussion by p-a-a-a-trick** [Eventual-Inc/Daft#1407] [Ideas]
Hey all, a colleague recently pushed a series of changes into Arrow adding asynchronous primitives to library device abstraction in Arrow C++ (latest: [Arrow#37365](https://github.com/apache/arrow/pull/37365)). I believe they're working on PyArrow wrappers in the near future and I'm trying to get feedback/use-cases for his team from end-users.

So I guess I'm wondering: is this something that the Daft project (or I guess any Eventual project) can benefit from? Are you able to share any particular insight into how this can be useful, or, better yet, are you able to try out these changes (either now or when they hit pyarrow) to provide feedback?

Sorry if this is the wrong space for this—let me know if it makes sense to post this somewhere else, or if it's not relevant to the Daft/Eventual-Inc projects at all that's fine too (just let me know). This is mostly for awareness/info gathering.

**Comment by samster25** [comment]
Hi @p-a-a-a-trick, We currently aren't leveraging Arrow C++ but the rust arrow2 crate. However, I have done quite a bit of CUDA in my last gig and am happy to give feedback for primitives if that's what you are looking for.

**Reply by p-a-a-a-trick** [reply]
Hey @samster25, sorry for the late reply. That'd be great. What's the best way to contact you?

Thanks!
